src/meltano/core/plugin/superset.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import subprocess
+from typing import List
+
+from meltano.core.behavior.hookable import hook
+from meltano.core.error import AsyncSubprocessError
+from meltano.core.plugin_invoker import PluginInvoker
+
+from . import BasePlugin, PluginType
+
+
+class SupersetInvoker(PluginInvoker):
+    def env(self):
+        env = super().env()
+
+        env["SUPERSET_HOME"] = str(self.plugin_config_service.run_dir)
+        env["SUPERSET_CONFIG_PATH"] = str(self.files["config"])
+        env["FLASK_APP"] = "superset"
+
+        return env
+
+
+class Superset(BasePlugin):
+    __plugin_type__ = PluginType.UTILITIES
+
+    invoker_class = SupersetInvoker
+
+    @property
+    def config_files(self):
+        return {"config": "superset_config.py"}
+
+    def process_config(self, flat_config):
+        # TODO: This means that `meltano config superset` doesn't show settings that are handled by Meltano itself, e.g. `ui.port` in the `ui` command
+        return {key: value for key, value in flat_config.items() if key.isupper()}
+
+    @hook("before_configure")
+    async def before_configure(self, invoker: SupersetInvoker, session):  # noqa: WPS217
+        """Write plugin configuration to superset_config.py."""
+        config_path = invoker.files["config"]
+        with open(config_path, "w") as config_file:
+            config = invoker.plugin_config_processed
+            config_file.write(
+                "\n".join(
+                    [
+                        "import sys",
+                        "module = sys.modules[__name__]",
+                        f"config = {str(config)}",
+                        "for key, value in config.items():",
+                        "    setattr(module, key, value)",
```

Comment:
This being a local temporary file with cleartext secrets creates several problems, as we've run into with Airflow. 

Why not have a files bundle that creates a fixed `config.py`:

```
import sys
module = sys.modules[__name__]

ENV_VAR_PREFIX = f"SUPERSET_"

meltano_config = {k.rstrip(ENV_VAR_PREFIX: v for os.environ.items() if k.startswith(ENV_VAR_PREFIX)}
for key, value in meltano_config.items():,
    setattr(module, key, value)
```

Reply:
@aaronsteers Interesting to hear you leaning towards the file bundle approach. I'm not actually aware of the Airflow problems you're talking about, and was modeling this approach after that one.

Reply:
@DouweM - I would prefer this be clear, explicit, and non-magic. If the file is in the repo, it can easily be packaged or volume-mapped into a docker container, and the user can optionally customize and control the behavior as needed.

The temporary file that deletes itself afterwards prevents alternative deployment options.

I'll try to get on a zoom to discuss the vision here.

Reply:
We decided that in this first iteration this approach that completely abstracts away configuration is acceptable, but it will not be sufficient for long, and we have to discuss more how prescriptive we want to be about how plugin configuration should be managed and how/when to leave the door open for users to do it the non-Meltano way.

Reply:
@DouweM - I'm okay with running the db upgrade as an action but I'd still prefer to move the `config.py` file back into the files bundle. Otherwise, we'll have a breaking change for all superset users when we do so later and this will be very difficult to unwind.

And as discussed, users who go to the docs expecting to have a config python file to modify will otherwise not have one.